src/test/java/org/mockitousage/stubbing/StrictStubbingEndToEndTest.java

```diff
@@ -93,6 +99,25 @@ public void run() {
         }
     }
 
+    public static class UnnecessaryStubbingWithImplicitEqMatchers {
+        @Mock IMethods mock;
+        MockitoSession mockito = Mockito.mockitoSession().initMocks(this).strictness(Strictness.STRICT_STUBS).startMocking();
+
+        @After public void after() {
+            mockito.finishMocking();
+        }
+
+        @Test public void unnecessary_stubbing() {
+            given(mock.simpleMethod("1")).willReturn("one");
```

Comment:
Those are neat tests! In addition, I'd love to see coverage for the following use case:

```
when(mock.foo(anyString()).thenReturn(x);
when(mock.foo("foo")).thenReturn(x);
```

In above, we also should report 2 unused stubs. This way, we cover the case with arg matcher.

Reply:
If the test contains, for example, the `mock.foo("bar");` call, the test should report only one unused stubbing (the one with the `"foo"` argument). What did you mean by 2 unused stubs in this case?

Reply:
I meant this:

```
when(mock.foo(anyString()).thenReturn(x);
when(mock.foo("foo")).thenReturn(x);
//2 unused stubbings
```

```
when(mock.foo(anyString()).thenReturn(x);
when(mock.foo("foo")).thenReturn(x);

mock.foo("foo")
//1 unused stubbing
```

Hope it makes sense!

Reply:
Got it.